File: lab6.cpp
#include <iostream>
using namespace std;

double equation(double n){

if(n==0){

return 1;

}
else{

return ((n/(n+2))-10)*equation(n-1);

}

}

double equation(){

cout<<"Please enter a number: "<<endl;
double x;
cin>>x;

if(x==1){
    return (x/(x+2))-10;
}
else{
    return ((x/(x+2))-10)*equation(x-1);
}
}

int main(){

cout<<"Please enter a number: "<<endl;
int n;
cin>>n;
cout<<equation(n)<<endl;
cout<<equation()<<endl;

}

File: lab6.py
#1
def equation1(n):
 f=lambda x:(x-3)/(x*x)
 list=[]
 for i in range(1,n+1):
    list.append(f(i))
 print(sum(list))

n=int(input("Please enter a number: "))
equation1(n)

#2
def equation2(n):
    """The function calculates and prints the result of the equation using a recursive function"""
    if n==0:
        return 1
    else:
        return (((n/(n+2))-10)*equation2(n-1))

n=int(input("Please enter a number: "))
print(equation2(n))
print(equation2.__doc__)





